xcache/xrootd/50-docker-paths.cfg

```
@@ -0,0 +1,11 @@
+# Make the caching plugin aware of the oss.space locations
+if nameed stash-origin
+  # do nothing
+else if named stash-origin-auth
+  # do nothing
+else
+  pfc.spaces data meta
+fi
+
+oss.space meta /xcache/
+oss.space data /xcache/disk*
```

This seems extremely fishy should not be this ones 
```suggestion
oss.space meta $cache_dir
oss.space data $cache_dir/disk*
```

---

But evne in this case I am not sure if I want to add all disks and at least on my deployments they are like

/data1/xcache 
/data2/xcache

Not the other way around

---

Nah, we'll be telling users to mount disks to:

1. `/xcache/meta`, `/xcache/disk1.../xcache/diskN`, and `/xcache/namespace` in the multi-disk scenario
2. or `/xcache/` in the single disk scenario

In the current implementation, there's no way for the user to specify the root cache dir (and we don't want to since we want to be prescriptive here).

---

I have the feeling this will break lots of stuff so $cache_dir is not localroot?

---

Nope, localroot is `/xcache/namespace`. I see your point about "breaking" existing caches since the symlinks for old disks won't point to the right places anymore, so I think deploying with these defaults will effectively clear your caches (fyi @ivukotic). Still, I think it'd be worth ripping off the bandaid now so we have more sensible container locations moving forward.

---

I am not gonna update all my yamls so I am now mounting /xcac/disk instead of /disk* just of this. Since these docker images are deployed in production all changes must be backward compatible. You can put an if around it. But forcing people to use this is not the way to go.

---

Where are your `oss.localroot`, `oss.space meta`, and `oss.space data` configs?

To transition, we'd just have to fixup symlinks on the partitions you're mounting and that shouldn't be very hard.

---

This is what I use for different cachees:
https://github.com/efajardo/prp-cms-xcache/blob/master/90-node-disks.cfg
https://github.com/efajardo/prp-cms-xcache/blob/master/90-node-xcache-11-disks.cfg

---

@abh3 can correct me if I'm wrong but I think your config should be fairly easy to transition to the new directory layout proposed in this PR:

1. Just mount whatever you're currently mounting to `/xcache-root` to `/xcache/namespace`
2. Modify the metadata symlinks in the host dir to that you're currently mounting to `/xcache-root` to point to `/xcache/meta/`
3. Modify the data symlinks in the host dir that you're currently mounting to `/xcache-root` to point to `/xcache/disk1.../xcache-diskN`

In addition to fixing symlink permissions, 2 and 3 can easily be done with a <10 line shell script. And if XRootD config overrides work as intended, the configuration that you have specified should work on top of the default config since in my experience, XRootD ignores missing paths for `oss.space` directives as long as there's one directive with a valid path. This way, you can transition to the default paths on your own timeline.

---

The last one I am not so sure about and I really do not want do re do the layout of my pods.

---

In previous versions of the PR, I had the following config:

```
oss.space data /xcache/disk/disk*
oss.space date /xcache/
```

And that definitely worked when I didn't have any `/xcache/disk/disk*` directories. I assume that `oss.space meta` would handle missing paths in the same way. I think I can make two minor tweaks so that it will work with your existing config, with the only change on your end would be moving the mount point of your host localroot from `/xcache-root` to `/xcache/namespace`